conda/plugins/subcommands/doctor/health_checks.py

```diff
@@ -175,6 +178,64 @@ def requests_ca_bundle_check(prefix: str, verbose: bool) -> None:
             )
 
 
+def consistent_env_check(prefix: str, verbose: bool) -> None:
+    # get the plugin manager from context
+    pm = context.plugin_manager
+
+    # get prefix data
+    pd = PrefixData(prefix)
+
+    # get virtual packages record
+    virtual_packages = {
+        record.name: record for record in pm.get_virtual_package_records()
+    }
+
+    # collect missing dependency issues in a list
+    issues = {}
+
+    # check if dependencies are present
+    for record in pd.iter_records():
+        for dependency in record.depends:
```

Comment:
The `record.depends` and `record.constraints` loops here are identical, we should combine them:

```
for dependency in (*record.depends, *record.constrains):
    ...
```

Reply:
The loops are not idential `constrains` does not require the package name to exist. The match is only performed if it a package with the matchspec name exists.

Reply:
Oh sorry, you're right